### PR TITLE
Support for Community Tickets

### PR DESCRIPTION
**Pull request by borkweb**
Related: 

* https://github.com/moderntribe/events-community-tickets/pull/7
* https://github.com/moderntribe/events-tickets-woo/pull/29
* https://github.com/moderntribe/events-community/pull/51

See: https://central.tri.be/issues/38511

### CODE REVIEW

File: src/Tribe/Tickets/Tickets.php [hunk 1]
```diff
@@ -245,7 +245,7 @@ function __construct() {
 			// Register all Tribe__Events__Tickets__Tickets api consumers
 			self::$active_modules[$this->className] = $this->pluginName;
 
-			add_filter( 'tribe_events_tickets_modules', 		 array( $this, 'modules' 					 )		  );
+			add_filter( 'tribe_events_tickets_modules', array( $this, 'modules' ) );
 			add_action( 'tribe_events_tickets_metabox_advanced', array( $this, 'do_metabox_advanced_options' ), 10, 2 );
 
 			// Admin AJAX actions for each provider
```
Comment (barryhughes) [inline src/Tribe/Tickets/Tickets.php]: Do we want to compliment the following with `wp_ajax_nopriv_*` hooks to accommodate anonymous submissions?

Comment (borkweb) [inline src/Tribe/Tickets/Tickets.php]: The current plan is to only allow ticket submission for authenticated users (there's some payment settings that Community Organizers will need to have in place for creating tickets).

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [list.php:64](https://github.com/moderntribe/the-events-calendar/blob/4a129721fd542f03037f9c8b852fcaecc4cbe582/src/admin-views/tickets/list.php#L64)

> via jenkins/codesniffer-phpcs

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not ')' | [list.php:64](https://github.com/moderntribe/the-events-calendar/blob/4a129721fd542f03037f9c8b852fcaecc4cbe582/src/admin-views/tickets/list.php#L64)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Admin/Organizer_Chooser_Meta_Box.php [hunk 2]
```diff
@@ -104,6 +104,10 @@ protected function single_organizer_dropdown( $organizer_id ) {
 	 * @return void
 	 */
 	protected function edit_organizer_link( $organizer_id ) {
+		$organizer_pto = get_post_type_object( Tribe__Events__Main::ORGANIZER_POST_TYPE );
+		if ( ! current_user_can( $organizer_pto->cap->create_posts ) ) {
```
Comment (bordoni) [inline src/Tribe/Admin/Organizer_Chooser_Meta_Box.php]: Can we make a check to see if the Cap is there before using it?

File: src/Tribe/Main.php [hunk 3]
```diff
@@ -1539,8 +1576,12 @@ public function saved_venues_dropdown( $current = null, $name = 'venue[VenueID]'
 				);
 			}
 			if ( $venues || $my_venues ) {
+				$venue_pto = get_post_type_object( self::VENUE_POST_TYPE );
 				echo '<select class="chosen venue-dropdown" name="' . esc_attr( $name ) . '" id="saved_venue">';
-				echo '<option value="0">' . esc_html( sprintf( __( 'Use New %s', 'the-events-calendar' ), $this->singular_venue_label ) ) . '</option>';
+
+				if ( current_user_can( $venue_pto->cap->create_posts ) ) {
```
Comment (bordoni) [inline src/Tribe/Main.php]: Same thing about the Cap usage.

File: src/Tribe/Main.php [hunk 4]
```diff
@@ -1621,8 +1662,12 @@ public function saved_organizers_dropdown( $current = null, $name = 'organizer[O
 				);
 			}
 			if ( $organizers || $my_organizers ) {
+				$oganizer_pto = get_post_type_object( self::ORGANIZER_POST_TYPE );
 				echo '<select class="chosen organizer-dropdown" name="' . esc_attr( $name ) . '" id="saved_organizer">';
-				echo '<option value="0">' . esc_html( sprintf( __( 'Use New %s', 'the-events-calendar' ), $this->singular_organizer_label ) ) . '</option>';
+
+				if ( current_user_can( $oganizer_pto->cap->create_posts ) ) {
```
Comment (bordoni) [inline src/Tribe/Main.php]: Check if the cap exists before using it?

File: src/Tribe/Main.php [hunk 5]
```diff
@@ -2956,8 +3001,11 @@ public function addEventMeta( $postId, $post ) {
 			 * When we have a VenueID/OrganizerID, we just save the ID, because we're not
 			 * editing the venue/organizer from within the event.
 			 */
+			$venue_pto = get_post_type_object( self::VENUE_POST_TYPE );
 			if ( isset( $_POST['Venue']['VenueID'] ) && ! empty( $_POST['Venue']['VenueID'] ) ) {
 				$_POST['Venue'] = array( 'VenueID' => intval( $_POST['Venue']['VenueID'] ) );
+			} elseif ( ! current_user_can( $venue_pto->cap->create_posts ) ) {
```
Comment (bordoni) [inline src/Tribe/Main.php]: Check if the cap exists before using it?

File: src/Tribe/Main.php [hunk 6]
```diff
@@ -2979,7 +3028,7 @@ public function normalize_organizer_submission( $submission ) {
 				foreach ( $submission['OrganizerID'] as $key => $organizer_id ) {
 					if ( ! empty( $organizer_id ) ) {
 						$organizers[] = array( 'OrganizerID' => intval( $organizer_id ) );
-					} else {
+					} elseif ( current_user_can( $organizer_pto->cap->create_posts ) ) {
```
Comment (bordoni) [inline src/Tribe/Main.php]: Check if the cap exists before using it?

File: src/Tribe/Tickets/Orders_Table.php [hunk 7]
```diff
@@ -0,0 +1,501 @@
...
+		return '';
+	}//end pagination
+
+	/**
+	 * Checks the current user's permissions
+	 *
+	 * @access public
+	 */
+	public function ajax_user_can() {
+		return current_user_can( get_post_type_object( $this->screen->post_type )->cap->edit_posts );
```
Comment (bordoni) [inline src/Tribe/Tickets/Orders_Table.php]: Check if the cap exists before using it?

File: src/admin-views/new-organizer-meta-section.php [hunk 8]
```diff
@@ -8,9 +8,12 @@
 	die( '-1' );
 }
 
+$organizer_pto = get_post_type_object( Tribe__Events__Main::ORGANIZER_POST_TYPE );
+
 ?>
 <script type="text/template" id="tmpl-tribe-create-organizer">
 <tbody class="new-organizer">
+<?php if ( current_user_can( $organizer_pto->cap->create_posts ) ) { ?>
```
Comment (bordoni) [inline src/admin-views/new-organizer-meta-section.php]: Check if the cap exists before using it?

File: src/admin-views/tickets/list.php [hunk 9]
```diff
@@ -27,18 +27,43 @@
...
 		}
 
-		$report = $provider_obj->get_ticket_reports_link( $post_id, $ticket->ID );
-		if ( $report ) {
-			$controls[] = $report;
+		if ( is_admin() ) {
+			if ( $ticket->admin_link ) {
+				$controls[] = sprintf( "<span><a href='%s'>" . __( 'Edit in %s', "tribe-events-calendar" ) . '</a></span>', esc_url( $ticket->admin_link ), $modules[$ticket->provider_class] );
+			}
+			if ( $ticket->frontend_link && get_post_status( $post_id ) == 'publish' ) {
```
Comment (bordoni) [inline src/admin-views/tickets/list.php]: `===`?

File: src/admin-views/tickets/list.php [hunk 10]
```diff
@@ -27,18 +27,43 @@
...
-					<?php echo $provider_obj->get_event_reports_link( $post_id ); ?>
+				<h4 class="tribe_sectionheader">
+					<?php
+					echo esc_html( apply_filters( 'tribe_events_tickets_module_name', $modules[$ticket->provider_class], $ticket->provider_class ) );
+					echo $provider_obj->get_event_reports_link( $post_id );
+					?>
 					<small>&nbsp;|&nbsp;</small>
-					<?php echo sprintf( '<small><a title="' . esc_attr__( 'See who purchased tickets to this event', 'the-events-calendar' ) . '" href="%s">%s</a></small>', esc_url( admin_url( sprintf( 'edit.php?post_type=%s&page=%s&event_id=%d', Tribe__Events__Main::POSTTYPE, Tribe__Events__Tickets__Tickets_Pro::$attendees_slug, $post_id ) ) ), __( 'Attendees', 'the-events-calendar' ) ); ?>
+					<?php
+					$attendees_url = admin_url( sprintf(
```
Comment (bordoni) [inline src/admin-views/tickets/list.php]: `add_query_arg`?

File: src/admin-views/tribe-options-timezones.php [hunk 11]
```diff
@@ -24,7 +24,7 @@
 return array(
 	'tribe_events_timezones_title' => array(
 		'type' => 'html',
-		'html' => '<h3>' . __( 'Timezone Settings', 'the-events-calendar' ) . '</h3>',
+		'html' => '<h2>' . __( 'Timezone Settings', 'the-events-calendar' ) . '</h2>',
```
Comment (bordoni) [inline src/admin-views/tribe-options-timezones.php]: `esc_html__`?

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:warning: | Expected next thing to be an escaping function (see Codex for 'Data Validation'), not '$this' | [Organizer_Chooser_Meta_Box.php:119](https://github.com/moderntribe/the-events-calendar/blob/f93148e2b42457f09b98a17a98fe0b199e8920cd/src/Tribe/Admin/Organizer_Chooser_Meta_Box.php#L119)

> via jenkins/codesniffer-phpcs

**Comment by bordoni** [comment]
:wink:

**Pull request merged**